docs/branching/new_branching_models.md

```diff
@@ -0,0 +1,111 @@
+
+### Example 1: Pre-merge hooks - enforce best practices
+
+Examples for good practices enforced in organizations:  
+
+ - No user_* columns except under /private/...
+ - Only `(*.parquet | *.orc | _delta_log/*.json)` files allowed
+ - Under /production, only backwards-compatible schema changes are allowed
+ - New tables on master must be registered in our metadata repository first, with owner and SLA
+
+lakeFS will assist in enforcing best practices by giving you a designated branch  to ingest new data (“new-data-1” in drawing). . You may run automated tests to validate  predefined  best practices as pre-merge hooks. If the validation passes, the new data will be automatically and atomically merged to the master branch. However, if the validation fails, you will be alerted, and the new data will not be exposed to consumers.
+
+By using this branching model and implementing best practices as pre merge hooks, you ensure the master lake is never compromised.
+
+<img src="../assets/img/branching_5.png" alt="branching_5" width="500px"/>
+
+
+## Continuous Deployment
+Not every day we introduce new data to the lake, or add/change ETLs, but we do have recurring jobs that are running, and updates to our existing data collections. Even if  the code and infra didn't change, the data might, and those changes introduce quality issues. This is one of the complexities of a data product, the data we consume changes over the course of a month, a week, or even a single day. 
+
+**Examples of changes to data that may occur:**
+ -  A client side bug in the data collection of website events
+ -  A new Android version that interferes with the collecting events from your App
+ -  COVID-19 abrupt impact on consumers behavior , and its effect on the accuracy of ML models.
+ - During a change to Salesforce interface, the validation requirement from a certain field had been lost
+
+lakeFS helps you validate your expectations and assumptions from the data itself.
+
+
+### Example 1: Pre merge hook - a data quality issue
+
+Continuous deployment of existing data we expect to consume, flowing from our ingest-pipelines into the lake. Similar to the Continuous Integration use-case   - we create an ingest branch (“events-data”), which allows us to create tests using data analysis tools or data quality services (e.g. [Great Expectations](https://greatexpectations.io/){: target="_blank" }, [Monte Carlo](https://www.montecarlodata.com/){: target="_blank" }) to ensure reliability of the data we merge to the master branch. Since merge is atomic, no performance issue will be introduced by using lakeFS, but your master branch will only include quality data. 
```

Comment:
```suggestion
Continuous deployment of existing data we expect to consume, flowing from our ingest-pipelines into the lake. Similar to the Continuous Integration use-case   - we create and ingest branch (“events-data”), which allows us to create tests using data analysis tools or data quality services (e.g. [Great Expectations](https://greatexpectations.io/){: target="_blank" }, [Monte Carlo](https://www.montecarlodata.com/){: target="_blank" }) to ensure reliability of the data we merge to the master branch. Since merge is atomic, no performance issue will be introduced by using lakeFS, but your master branch will only include quality data. 
```